### PR TITLE
Fixing the ordering of how we write long strings to the property store

### PR DESCRIPTION
**Pull request by mneedham**
* Need to write the dynamic part first otherwise we can end up with
  DynamicRecord Not in use type exceptions

### CODE REVIEW

**Comment by Davidmylove** [comment]
Good thing I can see it all take pics lol 2 bad I tried and tried you u get
her to stop for wrong me the alerts that's how we been getting everything
and it's been a while thank you chris for not letting me report it plus I
got the iP address from work I was never rude or disrespectful but when I
get pushed fuck it she shouldn't been cheating and still sneaking around I
stop contact with her when I found out we got more then enough I tried 1st
to let her know take me off am done but no she need to fight to feel love
well I need to feel loved not fight so sorry but I gotta do what I gotta
do.

File: community/kernel/src/test/java/org/neo4j/kernel/impl/store/PropertyStoreTest.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
+package org.neo4j.kernel.impl.store;
```
Comment (thobe) [inline community/kernel/src/test/java/org/neo4j/kernel/impl/store/PropertyStoreTest.java]: Missing license header...

Comment (mneedham) [inline community/kernel/src/test/java/org/neo4j/kernel/impl/store/PropertyStoreTest.java]: D'oh, sorry! Have put the header in now hopefully.

File: community/kernel/src/test/java/org/neo4j/kernel/impl/store/PropertyStoreTest.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+        DynamicRecord dynamicRecord = dynamicRecord();
+        PropertyBlock propertyBlock = propertyBlockWith( dynamicRecord );
+        record.setPropertyBlock( propertyBlock );
+
+        // when
+        store.updateRecord( record );
+
+        // then
+        InOrderImpl inOrder = new InOrderImpl( Arrays.<Object>asList(stringPropertyStore, cursor) );
```
Comment (thobe) [inline community/kernel/src/test/java/org/neo4j/kernel/impl/store/PropertyStoreTest.java]: Use `Mockito.inOrder( stringPropertyStore, cursor )` instead.

**Comment by thobe** [comment]
Would have been nice to have a test for the actual issue as well, i.e. testing that concurrent readers do not see incomplete properties...

**Comment by Davidmylove** [comment]
Good influence no wonder you people brake into phone email accounts and make pronos at work

**Comment by mneedham** [comment]
Where should that type of test go? The way I tested the issue was to run
the ReadLongStringBenchmark a few times. That fails all the time without
this fix.

**Comment by thobe** [comment]
@mneedham it would go somewhere in the rest of the mess of tests, as far as I know there is no structure to where tests go.

**Comment by mneedham** [comment]
Ok. Have added an integration test which makes the exception happen every
few hundred iterations of the loop. Reordering the logic fixes it -
https://github.com/mneedham/neo4j/blob/b652f148b6495b2f0c08087eca535578b50a5ce3/community/kernel/src/test/java/org/neo4j/kernel/impl/store/PropertyStoreIT.java

File: community/kernel/src/test/java/org/neo4j/kernel/impl/store/PropertyStoreIT.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.store;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
```
Comment (tinwelint) [inline community/kernel/src/test/java/org/neo4j/kernel/impl/store/PropertyStoreIT.java]: Please clean up unused imports. This class has many unused.

**Comment by tinwelint** [comment]
The test failure is HaLoggingIT. A rebase of this PR off of master will solve that, if you'd like to have a green build.

**Comment by tinwelint** [comment]
@mneedham The test failure is HaLoggingIT. A rebase of this PR off of master will solve that, if you'd like to have a green build.

**Pull request merged**